### ISSUE TITLE
Heatmap reorders DOM elements on hover

### ISSUE THREAD

**Issue by oysteinmoseng** [highcharts/highcharts#13049]
Heatmap series reorders DOM elements as they are hovered, in order to push them to the top of the z-index. This is presumably done in order to ensure that their border is visible, in case they have a different border style on hover.

This causes some a11y issues with screen readers, as the ordering is dynamically altered.

Could we consider only pushing the points to the top if they actually have a border style on the hover state?

The code in question can be found in `Highcharts.colorMapPointMixin.setState`.

Workaround: https://jsfiddle.net/oysteinmoseng/hm3efjwk/1/

#### Live demo with steps to reproduce
https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/demo/heatmap/

**Comment by pawelfus** [comment]
Perhaps a11y can connect to `Highcharts.colorMapPointMixin.setState` and reorder a11y elements?

I think hover state with custom border might be quite popular in heatmap  (https://jsfiddle.net/cm6kd1wf/) because of increased readability.

Alternatively, w can create a halo-element in heatmap that will be rendered on top of the hovered cell. It's also not a perfect solution, but then we can resign from bringing up to front the cell. This would be a small breaking change, but can be included in changelog.

**Comment by oysteinmoseng** [comment]
We could always force the order in a11y (like in the workaround), but if it breaks the z-indexing it's not really improving things much.

I like the idea of a halo. It doesn't even have to be a breaking change as long as it doesn't use halo options - or am I missing something? Or we could add it to the changelog as you mention. Tilemap already supports halos, and it should be a simple implementation if we use the standard halo options.

Thoughts @TorsteinHonsi?

**Comment by raf18seb** [comment]
@oysteinmoseng I'm not able to reproduce it (in current 9.3.2 and 8.0.4 from March 2020). Could you double-check? Maybe it's been solved and the ticket can be closed.

**Comment by oysteinmoseng** [comment]
Still reorders for me, the point `<path>` that is hovered is moved last in the DOM order: [demo]

See the last element in the SVG group changing:

(https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/demo/heatmap/).
<img width="1046" alt="Screen Shot 2021-12-03 at 10 08 59" src="https://user-images.githubusercontent.com/5276642/144576233-05fd88e0-5a30-477b-9814-77d078ee5efa.png">
<img width="1024" alt="Screen Shot 2021-12-03 at 10 08 44" src="https://user-images.githubusercontent.com/5276642/144576254-ea7c7ab4-80ed-40f4-817c-4ca9681b30c0.png">

**Comment by pandya22** [comment]
Hello all,

The workaround suggested throws us compilation errors. We are using Highcharts version - 10.3.3. Could you please help us with a workaround that is compatible with our version?

![image](https://github.com/highcharts/highcharts/assets/143558048/a884eba6-e4d0-4ccc-b381-e6a9f1817fcb)

**Comment by hubertkozik** [comment]
Hi @pandya22!

I've just checked it and the workaround works correctly for Highcharts v10.3.3: https://jsfiddle.net/BlackLabel/L06ep4yv/

We use GitHub as a tool for bugs/feature request tracking, so if you need further help with the configuration of your chart, feel free to reach out to our other [support channels](https://www.highcharts.com/blog/support/), where our support engineers will help you with the configuration of your chart.

**Comment by TorsteinHonsi** [comment]
### Workaround
I found another workaround that not only disables reordering of elements, but also retains the visuals of the border being on top. It works by creating a `use` element to mirror the hovered element. 

This should be considered as a permanent fix for this issue, both in heatmaps and in geo maps.

Please check this @pandya22 and see if it works for you:

https://jsfiddle.net/highcharts/y6gr1bch/

**Comment by pandya22** [comment]
Hello all,

Unfortunately, these solutions do now work for us. Most likely because we have a very large dataset. Here is the fiddle - https://jsfiddle.net/payPan22/hLtbxv3w/10/  where you can see the issue.

In the attached gif, look at the bottom right blocks (x ~ 35), there are 2 blocks originally, and after hovering they disappear.
![HighchartsHeatmap_2 (1)](https://github.com/highcharts/highcharts/assets/143558048/bb29fb34-6aa8-4f9d-b39d-a69cd9d6d022)

**Comment by hubertkozik** [comment]
@pandya22, can you try the latest workaround provided above? I've prepared a demo based on yours: https://jsfiddle.net/BlackLabel/hwte632m/ and everything looks correct.

**Comment by yugako** [comment]
@hubertkozik The issue is still here, even on your fiddle.
![chrome-capture-2024-3-22 (2)](https://github.com/highcharts/highcharts/assets/27961118/0a33da02-2968-498c-a117-9cb9a6b2d41f)

**Comment by hubertkozik** [comment]
@yugako I don't see any reorders in the demo from my previous message and on your gif. The pinkish point is behind the navy points and after hovering it shows up correctly. For example, we can change the width of the chart to something really big: https://jsfiddle.net/BlackLabel/3txmr7L4/ and see points more precisely. You can check even in the console, that point is not moved/reordered in the DOM structure after hovering.
